weeks/week6/report.md

```diff
@@ -12,85 +12,91 @@
 
 ### Problem 1: Floating point numbers
 ```
-insert your pattern here
+(\d)\.\d+
 ```
 ### Problem 2: Years before 1990
 ```
-insert your pattern here
+.+.19[1-8][0-9].
 ```
 ### Problem 3: Hexadecimal colors
 ```
-insert your pattern here
+#(\w|\d){6}
 ```
 ### Problem 4: Grayscale colors
 ```
-insert your pattern here
+#([a-z0-9]{2}|[A-Z0-9])\1{2,}
 ```
 ### Problem 5: Too long lines
 ```
-insert your pattern here
+.{30,}
 ```
 ### Problem 6: Remove repeating words
 ```
-insert your pattern here
+(\s[\w-']+)\1
+$1
 ```
 ### Problem 7: Match HTML tags
 ```
-insert your pattern here
+<[!/]?\w+\s?(\w+)?>
```

Comment:
Good one here as well, there are certainly many options to do this, just am gonna mention that `(\w+)?` is just `\w*`, right?